tests/stress/consume_flow_with_threads_test.py

```diff
@@ -0,0 +1,164 @@
+    file3_url = (
+        "https://dumps.wikimedia.org/enwiki/latest/enwiki-latest-abstract10.xml.gz"
+    )
+    file3_dict = {"type": "url", "url": file3_url, "method": "GET"}
+    file3 = FilesTypeFactory(file3_dict)
+    files = [file1, file2, file3]
+
+    erc721_nft, erc20_token = deploy_erc721_erc20(ocean.web3, config, wallet, wallet)
+    encrypt_response = data_provider.encrypt(
+        [random.choice(files)], config.provider_url
+    )
+    encrypted_files = encrypt_response.content.decode("utf-8")
+
+    ddo = ocean.assets.create(
+        metadata=metadata,
+        publisher_wallet=wallet,
+        encrypted_files=encrypted_files,
+        erc721_address=erc721_nft.address,
+        erc20_templates=[1],
+        erc20_names=["Datatoken 1"],
+        erc20_symbols=["DT1"],
+        erc20_minters=[wallet.address],
+        erc20_fee_managers=[wallet.address],
+        erc20_publish_market_order_fee_addresses=[ZERO_ADDRESS],
+        erc20_publish_market_order_fee_tokens=[get_address_of_type(config, "Ocean")],
+        erc20_caps=[ocean.to_wei("0.5")],
+        erc20_publish_market_order_fee_amounts=[0],
+        erc20_bytess=[[b""]],
+    )
+
+    assert ddo, "The asset is not created."
+    assert ddo.nft["name"] == "NFT"
+    assert ddo.nft["symbol"] == "NFTSYMBOL"
+    assert ddo.nft["address"] == erc721_nft.address
+    assert ddo.nft["owner"] == wallet.address
+    assert ddo.datatokens[0]["name"] == "Datatoken 1"
+    assert ddo.datatokens[0]["symbol"] == "DT1"
+    assert ddo.credentials == build_credentials_dict()
+
+    service = get_first_service_by_type(ddo, ServiceTypes.ASSET_ACCESS)
+
+    # Initialize service
+    response = data_provider.initialize(
+        did=ddo.did, service=service, consumer_address=wallet.address
+    )
+    assert response
+    assert response.status_code == 200
+    assert response.json()["providerFee"]
+
+    erc20_token.mint(
+        account_address=wallet.address,
+        value=ocean.to_wei(20),
+        from_wallet=wallet,
+    )
+    # Start order for consumer
+    tx_id = erc20_token.start_order(
+        consumer=wallet.address,
+        service_index=ddo.get_index_of_service(service),
+        provider_fee_address=response.json()["providerFee"]["providerFeeAddress"],
+        provider_fee_token=response.json()["providerFee"]["providerFeeToken"],
+        provider_fee_amount=response.json()["providerFee"]["providerFeeAmount"],
+        v=response.json()["providerFee"]["v"],
+        r=response.json()["providerFee"]["r"],
+        s=response.json()["providerFee"]["s"],
+        valid_until=response.json()["providerFee"]["validUntil"],
+        provider_data=response.json()["providerFee"]["providerData"],
+        consume_market_order_fee_address=wallet.address,
+        consume_market_order_fee_token=erc20_token.address,
+        consume_market_order_fee_amount=0,
+        from_wallet=wallet,
+    )
+    # Download file
+    destination = config.downloads_path
```

Comment:
can we move 121-131 into a function? It seems useful, maybe even for the main functionality. Also, are you sure this works on both mac and linux? AFAIK, there are some tiny differences with os dir utils.

Reply:
I haven't tested on mac, just linux environment.